app/src/main/java/com/sopt/instagram/ui/home/HomeFragment.kt

```diff
@@ -21,8 +25,23 @@ class HomeFragment : BindingFragment<FragmentHomeBinding>(R.layout.fragment_home
     }
 
     private fun initAdapter() {
-        binding.rvHomePost.adapter = PostAdapter().apply {
-            submitList(viewModel.mockPostList)
+        viewModel.getPostList()
+        viewModel.getPostListState.observe(viewLifecycleOwner) { state ->
+            when (state) {
+                is UiState.Success -> {
+                    binding.rvHomePost.adapter = PostAdapter().apply {
+                        submitList(viewModel.postList.value)
+                    }
+                }
+
+                is UiState.Failure -> {
+                    requireContext().showToast("erererer")
+                }
+
+                else -> {
+                    // nothing
+                }
+            }
         }
```

Comment:
```suggestion
        viewModel.getPostListState.observe(viewLifecycleOwner) { state ->
            when (state) {
                is UiState.Success -> {
                    binding.rvHomePost.adapter = PostAdapter().apply {
                        submitList(viewModel.postList.value)
                    }
                }

                is UiState.Failure -> {
                    requireContext().showToast("erererer")
                }

                is UiState.Error -> {
                    // nothing
                }
            }
        }
```

sealed class의 이점을 제대로 활용하기 위해서는 esle 구문 보다는 UiState에 포함되어있는 분기들을 모두 작성하는 것이 좋을 것 같습니다 :)

Reply:
넵!! 수정하겠습니다!